linalg/complex_operator.cpp

```diff
@@ -13,6 +13,26 @@
 #include <set>
 #include <map>
 
+// Define macro wrappers for hypre_TAlloc, hypre_CTAlloc and hypre_TFree:
+// mfem_hypre_TAlloc, mfem_hypre_CTAlloc, and mfem_hypre_TFree, respectively.
+// Note: the same macros are defined in hypre.cpp and hypre_parser.cpp.
+#if MFEM_HYPRE_VERSION < 21400
+
+#define mfem_hypre_TAlloc(type, size) hypre_TAlloc(type, size)
+#define mfem_hypre_CTAlloc(type, size) hypre_CTAlloc(type, size)
+#define mfem_hypre_TFree(ptr) hypre_TFree(ptr)
+
+#else // MFEM_HYPRE_VERSION >= 21400
+
+// See the notes about hypre 2.14.0 in hypre.cpp
+#define mfem_hypre_TAlloc(type, size) \
+   hypre_TAlloc(type, size, HYPRE_MEMORY_HOST)
+#define mfem_hypre_CTAlloc(type, size) \
+   hypre_CTAlloc(type, size, HYPRE_MEMORY_HOST)
+#define mfem_hypre_TFree(ptr) hypre_TFree(ptr, HYPRE_MEMORY_HOST)
+
+#endif // #if MFEM_HYPRE_VERSION < 21400
+
```

Comment:
Instead of copy+pasting this from `hypre.cpp`, it seems better to move them to `hypre.hpp`.

Reply:
I do agree but I assumed someone had a good reason for duplicating this code block in `hyper_parser.cpp`.  If not, perhaps we should fix that as a separate PR since it's a bit off on a tangent.  I don't mind doing that and it would give the author the chance to make their case.

Reply:
I just added a separate PR #1172 to address this.